### PR TITLE
[stable/datadog] Security Agent settings for compliance and runtime security

### PR DESCRIPTION
**Pull request by xornivore**
#### What this PR does / why we need it:

Changes to enable running security agent container for compliance monitoring and runtime security.

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [ ] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [ ] Chart Version bumped
- [ ] Variables are documented in the README.md
- [ ] Title of the PR starts with chart name (e.g. `[stable/mychartname]`)

### CODE REVIEW

File: stable/datadog/README.md [hunk 1]
```diff
@@ -328,6 +328,10 @@ helm install --name <RELEASE_NAME> \
 | `datadog.systemProbe.enableConntrack`                        | If true, system-probe connects to the netlink/conntrack subsystem to add NAT information to connection data. Ref: http://conntrack-tools.netfilter.org/                             | `true`                                          |
 | `datadog.systemProbe.bpfDebug`                               | If true, system-probe writes debug logs to /sys/kernel/debug/tracing/trace_pipe                                                                                                     | `false`                                         |
 | `datadog.systemProbe.apparmor`                               | Apparmor profile for system-probe                                                                                                                                                   | `unconfined`                                    |
+| `datadog.securityAgent.enabled`                              | Enable security agent                                                                                                                                                               | `false`                                         |
```
Comment (vboulineau) [inline stable/datadog/README.md]: Should not exist, can be deduced from `.compliance.enabled || .security.enabled` I guess

File: stable/datadog/templates/agent-rbac.yaml [hunk 2]
```diff
@@ -18,6 +18,7 @@ rules:
   - pods
   - nodes
   - componentstatuses
+  - podsecuritypolicies
```
Comment (vboulineau) [inline stable/datadog/templates/agent-rbac.yaml:28]: `podsecuritypolicies` belongs to the `policy` group

Comment (xornivore) [reply stable/datadog/templates/agent-rbac.yaml:28]: Ah good point, thank you!

File: stable/datadog/README.md (hunk 1 above)
Comment (xornivore) [reply stable/datadog/README.md]: This is following the approach similar to `processAgent` - where the container is enabled and then features are configured for it.

File: stable/datadog/templates/container-security-agent.yaml [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    {{- if eq .Values.targetSystem "linux" }}
+    - name: cgroups
+      mountPath: /host/sys/fs/cgroup
+      readOnly: true
+    - name: passwd
+      mountPath: /etc/passwd
+    - name: group
+      mountPath: /etc/group
+    - name: hostroot
+      mountPath: /host/root
```
Comment (ahmed-mez) [inline stable/datadog/templates/container-security-agent.yaml]: Let's add `readOnly: true`

**Comment by xornivore** [comment]
PR migrated [here](https://github.com/DataDog/helm-charts/pull/5)

**Pull request closed without merging**